nemo_curator/modules/fuzzy_dedup.py

```diff
@@ -1144,16 +1145,22 @@ def _batched_merge_and_write(
                 try:
                     # NOTE: If we have more text-df partitions than bucket-map
                     # partitions, we are more likely to see an OverflowError
+
+                    subset_merged_df = merge_left_to_shuffled_right(
+                        subset_text_df,
+                        subset_bucket_df,
+                        merge_on,
+                    )
+
                     output_df = text_bytes_aware_shuffle(
-                        df=merge_left_to_shuffled_right(
-                            subset_text_df,
-                            subset_bucket_df,
-                            merge_on,
-                        ),
+                        df=subset_merged_df,
                         partition_on=partition_on,
                         text_column=self.text_field,
                         num_workers=num_workers,
                     )
+                    if output_df is None:
+                        text_part_offset += parts_per_text_batch_use
```

Comment:
Reviewer please TAL at this

Reply:
I think the logic makes sense here to increment the `text_part_offset`. It might also make sense to call `update_restart_offset` here to update the txt file to restart from the next chunk if this job is resumed.

Reply:
Removed the continue logic based on @VibhuJawa 's suggestion so this isn't needed anymore. Thanks for catching that there is more to the control flow than what I caught